pkg/execution/svccatadmission/internaldns.go

```diff
@@ -0,0 +1,103 @@
+package svccatadmission
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/atlassian/voyager/pkg/microsserver"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/internaldns"
```

Comment:
I believe the current practice is to not reference across layers like this. At least, that's not how the other entries in svccatadmission work (e.g. https://github.com/atlassian/voyager/blob/master/pkg/execution/svccatadmission/micros.go#L92). They just validate the plan/service and then have the minimal json to match the expected spec, rather than referencing the whole spec across layers.

If this was discussed in person, please ignore.

Reply:
I think we should change the practice to:
- export the types for the resource type into a subpackage of its autowiring package
- use types from that package.

I have done that change in a very few places already. See `pkg/orchestration/wiring/k8scompute/api` as an example.
We can and should benefit from the fact that we have monorepo but we should not turn our build into "let everything depend on everything and then the linker will prune unused stuff". This would be bad for build speed.